### ISSUE TITLE
SSG (Super Simple Gimbal) Elevon Servo Mix Camera Stabilization. 

### ISSUE THREAD

**Issue by Homertje** [MegaPirateNG/ardupilot-mpng#46]
On the new(er) MultiWii 2.1 and up there's a great working code for simple and cheap camera stabilization. 
http://www.rcgroups.com/forums/showthread.php?t=1793759&page=9

The code should be quite simple: 
It would be the same code one would use to stabilize a delta-wing aircraft, since that code is implemented in arduplane to stabilize elevons, It should be a mere copy paste and change inputs/outputs. 

(elevon mixing code can be found @ arduplane/attitude.pde line 469 and down)

I/we really want that code to be running to camera gimbal servo outputs. (and camera roll/tilt as input) 

In short: my question is, can there be a additional camera stabilization mode, with the code behind elevon mixing stabilization?

**Comment by Homertje** [comment]
apperently the code could be added by changing "AP_Mount.cpp": 

The servo out put should then be: 
// write the results to the servos
    move_servo(_roll_idx, (_tilt_angle+_roll_angle)_10, _roll_angle_min_0.1f, _roll_angle_max_0.1f);
    move_servo(_tilt_idx, (_tilt_angle-_roll_angle)_10, _tilt_angle_min_0.1f, _tilt_angle_max_0.1f);
    move_servo(_pan_idx,  _pan_angle_10,  _pan_angle_min_0.1f,  _pan_angle_max*0.1f);

this is off DIY drones from a user named:  Rikard Andersson

**Comment by Homertje** [comment]
tested this code and it kind of works (code works flawlessly but cannot connect servos to a lot of pins, camera pins do nothing, rc10 spazzes servo all over the place, rc9, rc8 and rc6 does nothing), will open new issue for the new problem at hand.

Closing this issue, it would be nice when the camera stab pins work to add the code below as a option, (#define SSG_Mix ENABLED or something similar) suppose a simple code like this would work: 
somewhere in AP_Mount.cpp

```
#ifndef SSG_Mix 
# define SSG_Mix DISABLED 
#endif  
```

in void AP_Mount::update_mount_position():

```
#if SSG_Mix == ENABLED
    move_servo(_roll_idx, (_tilt_angle+_roll_angle)*10, _roll_angle_min*0.1f, _roll_angle_max*0.1f);
    move_servo(_tilt_idx, (_tilt_angle-_roll_angle)*10, _tilt_angle_min*0.1f, _tilt_angle_max*0.1f);
    move_servo(_pan_idx,  _pan_angle*10,  _pan_angle_min*0.1f,  _pan_angle_max*0.1f);
#else
    move_servo(_roll_idx, _roll_angle*10, _roll_angle_min*0.1, _roll_angle_max*0.1);
    move_servo(_tilt_idx, _tilt_angle*10, _tilt_angle_min*0.1, _tilt_angle_max*0.1);
    move_servo(_pan_idx,  _pan_angle*10,  _pan_angle_min*0.1,  _pan_angle_max*0.1); 
#endif
```

`#define SSG_Mix ENABLED`  (somewhere in AP_Mount.cpp or preferably APM_Config.h if possible) would enable mixed servo output, anything else would disable mixed output. Not defining would disable it aswell (eliminating a "not defined" error)

I'm aware this IF statement might use resources, but so far i think this would be a feasible option. Will test that code later today. in case the IF statement uses too much resources i'll try copying the entire void loop and rename it to AP_Mount::update_mount_position_SSG():

And maybe find out where it's called and make a if statement there to select which void loop to use.
but that's for later concern. 

I want to try offloading the cam stabilization to another arduino board, however for me it's too complicated (at this moment) to forward the MPU6050 data from the FC to a output. (i want to make sure I save money, space and weight by not adding another breakout gyro/acc board.)
If anyone knows how to forward the output of the MPU6050 to a digital pin or even a serial/I2C pin then i would be grateful.
